src/mds/LogSegment.h

```diff
@@ -34,10 +35,10 @@ class CDentry;
 class MDSRank;
 struct MDPeerUpdate;
 
-class LogSegment {
- public:
```

Comment:
`public:` starts a class. See:

https://google.github.io/styleguide/cppguide.html#Declaration_Order

Reply:
It says
> A class definition should **usually** start with a public: section, ...
(the emphasis is mine)

Also, below it requests that constructors precede other members. 
So, I'm not convinced this violates the requirement, but I will move it to the end if you think I should, just confirm please.

Reply:
Within `mds/` I took that "request" as policy because the status-quo was a complete mess of public/private back-n-forth. Cleaning that up used valuable dev time so I'm a stickler about not regressing.  See the commit history of the header files if interested.

Reply:
done